### PR TITLE
Bcf.gz to bcf (and reverse) converters + bcf_bgzip data type

### PR DESCRIPTION
**Pull request by markiskander**
Hey guys. I added a bcf_bgzip datatype as well as converters for getting it converted to bcf (and from bcf to bcf_bgzip). I did my best to closely follow the existing naming schemes and coding styles. Let me know if there's anything else that needs to be done. Thanks!

### CODE REVIEW

File: lib/galaxy/datatypes/converters/bcf_bgzip_to_bcf_converter.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import subprocess
+
+
+def main():
+    # Read options, args.
+    parser = optparse.OptionParser()
+    (options, args) = parser.parse_args()
+    input_fname, output_fname = args
+
+    subprocess.call(["bcftools", "view", input_fname, "-o", output_fname, "-O", "u"])
```
Comment (bgruening) [inline lib/galaxy/datatypes/converters/bcf_bgzip_to_bcf_converter.py:18]: @markiskander can you please move this command directly into the tools. This way we completely save this file :)

File: lib/galaxy/datatypes/converters/bcf_to_bcf_bgzip_converter.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<tool id="CONVERTER_bcf_to_bcf_bgzip" name="Convert BCF to BCF_BGZIP" version="0.0.1" hidden="false">
+<!--  <description>__NOT_USED_CURRENTLY_FOR_CONVERTERS__</description> -->
+  <command interpreter="python">bcf_to_bcf_bgzip_converter.py '$input1' '$output1'</command>
```
Comment (bgruening) [inline lib/galaxy/datatypes/converters/bcf_to_bcf_bgzip_converter.xml:3]: same here: you could write something like `bcftools view "$input1" -o "$output1" -O b`

**Comment by bgruening** [comment]
Thanks @markiskander looks great!

**Comment by jmchilton** [comment]
Thanks for the contribution @markiskander, I'm marking this as a "work in progress" until @bgruening's issues are addressed. Hope this is okay.

**Comment by ericenns** [comment]
Note @markiskander has finished his term and wont have time to work on this anymore.

**Comment by jmchilton** [comment]
@bgruening Given @markiskander isn't available to modify these tools, can we just merge as is? This won't be the only converter with an unnecessary wrapper?

**Comment by markiskander** [comment]
AFAIK this follows the same general template as vcf_to_vcf_bgzip converter, which I had used as a reference for this. It's practically the same thing but with bcfs instead.

**Comment by bgruening** [comment]
I will fix my comments post merge, no problem!
@markiskander thanks a lot for your contribution. Hopefully we will see you here again at some point! :)

**Pull request merged**